src/lib.rs

```diff
@@ -146,19 +146,22 @@ pub fn eval_beat(cmds: &[Cmd], t: f64) -> Result<f64, ()> {
                 }
             }
             Arr(size) => {
-                // We need to pop index + size values, so our stack needs to be
-                // atleast size + 1 elements long
-                if size + 1 > stack.len() || size == 0 { 
-                    stack.push(0.0)
+                let index = stack.pop().ok_or(())? as i32;
```

Comment:
Why use `i32` instead of `i64`?

~Alternately, does `as i64 as usize` give us the right behavior? `-1.0 as i64 as usize` gives `18446744073709551615` for me, which looks right. If this works nicely it would make some of the later casting unnecessary.~
Using `as i64 as usize` would give the wrong result for negative indices. `18446744073709551615 mod 3 = 0`, `-1 mod 3 = 2`.